packages/osd-stylelint-plugin-stylelint/src/rules/no_modifying_global_selectors/index.ts

```diff
@@ -40,15 +40,16 @@ const ruleFunction = (
 
     const isAutoFixing = Boolean(context.fix);
 
-    postcssRoot.walkRules((rule: any) => {
+    postcssRoot.walkRules((rule) => {
       const selectorRule = getSelectorRule(rules, rule);
       if (!selectorRule) {
         return;
       }
 
       let shouldReport = false;
 
-      const file = postcssRoot.source.input.file;
+      // We can safely unwrap these possibly undefined variables because we get these values from Stylelint, which grabs them from source files.
+      const file = postcssRoot.source!.input.file!;
```

Comment:
Maybe these should be explicitly checked? I feel a little hesitant to add hard unwraps like this

Reply:
Yeah, the generally better way would be optchaing them and then varifying that `file` defined, even if you know it should be unreachable.